test/unit/controllers/webhook.test.js

```diff
@@ -0,0 +1,751 @@
+  return {
+    create: mockCreateFn
+  }
+})
+
+const mockHmacDigest = 'mock Hmac digest'
+/*
+ * Mock the createHmac function within the native crypto module, but leave every other function.
+ * This allows us to test logic that invokes GitHub's webhook request authentication, which
+ * involves signing the webhook payload using the agreed-upon secret token.
+ */
+jest.mock('crypto', () => {
+  const cryptoOrig = require.requireActual('crypto')
+  return {
+    ...cryptoOrig,
+    createHmac: mockCreateHmacFn
+  }
+})
+
+jest.mock('../../../lib/Staticman', () => {
+  return jest.fn().mockImplementation(() => {
+    return {
+      setConfigPath: mockSetConfigPathFn,
+      getSiteConfig: mockGetSiteConfigFn,
+      processMerge: mockProcessMergeFn
+    }
+  })
+})
+
+// Instantiate the module being tested AFTER mocking dependendent modules above.
+const webhook = require('./../../../controllers/webhook')
+
+beforeEach(() => {
+  req = helpers.getMockRequest()
+  res = helpers.getMockResponse()
+
+  mockCreateFn.mockImplementation((service, options) => {
+    return {
+      getReview: mockGetReviewFn,
+      deleteBranch: mockDeleteBranchFn
+    }
+  })
+
+  mockCreateHmacFn.mockImplementation((algo, data) => {
+    return {
+      update: data => {
+        return {
+          digest: encoding => {
+            return mockHmacDigest
+          }
+        }
+      }
+    }
+  })
+})
+
+afterEach(() => {
+  mockGetReviewFn.mockClear()
+  mockDeleteBranchFn.mockClear()
+  mockCreateFn.mockClear()
+  mockSetConfigPathFn.mockClear()
+  mockGetSiteConfigFn.mockClear()
+  mockProcessMergeFn.mockClear()
+  mockCreateHmacFn.mockClear()
+})
+
+describe('Webhook controller', () => {
+  test.each([
+    ['gitfoo']
+  ])('abort and return an error if unknown service specified - %s', async (service) => {
+    req.params.service = service
+
+    expect.hasAssertions()
+    return webhook(req, res).then(response => {
+      expect(mockCreateFn).toHaveBeenCalledTimes(0)
+      expect(res.send.mock.calls[0][0]).toEqual({ errors: '[\"Unexpected service specified.\"]' })
+      expect(res.status.mock.calls[0][0]).toBe(400)
+    })
+  })
+
+  test.each([
+    ['github'], ['gitlab']
+  ])('abort and return an error if no event header found - %s', async (service) => {
+    req.params.service = service
+
+    expect.hasAssertions()
+    return webhook(req, res).then(response => {
+      expect(mockCreateFn).toHaveBeenCalledTimes(0)
+      expect(res.send.mock.calls[0][0]).toEqual({ errors: '[\"No event found in the request\"]' })
+      expect(res.status.mock.calls[0][0]).toBe(400)
+    })
+  })
+
+  test.each([
+    ['github'], ['gitlab']
+  ])('abort and return success if not "Merge Request Hook" event - %s', async (service) => {
+    req.params.service = service
+    if (service === 'github') {
+      req.headers['x-github-event'] = 'Some Other Hook'
+    } else if (service === 'gitlab') {
+      req.headers['x-gitlab-event'] = 'Some Other Hook'
+    }
```

Comment:
```suggestion
    req.headers[`x-${service}-event`] = 'Some Other Hook'
```

Can use fstrings here